### ISSUE TITLE
chunksize how to set

### ISSUE THREAD

**Issue by WuYanZhao107** [FullStackRetrieval-com/RetrievalTutorials#1]
![image](https://github.com/FullStackRetrieval-com/RetrievalTutorials/assets/113891875/c20c4cb9-0108-4e8d-be2c-20d6221f538b)
how can I set every chunksize?
I find it's too long for my llm

**Comment by jnash10** [comment]
ctrl+click on SemanticChunker

Reduce `breakpoint_percentile_threshold=95` on line 103

**Comment by WuYanZhao107** [comment]
you mean I just change the lower the number of `breakpoint_percentile_threshold = 95`?
but I test lower to 50, I find I still have 36 chunks. 
and a funny thing is the second chunk from third page of my pdf to the Penultimate page

**Comment by WuYanZhao107** [comment]
> ctrl+click on SemanticChunker
> 
> Reduce `breakpoint_percentile_threshold=95` on line 103

you mean I just change the lower the number of breakpoint_percentile_threshold = 95?
but I test lower to 50, I find I still have 36 chunks.
and a funny thing is the second chunk from third page of my pdf to the Penultimate page

**Comment by jnash10** [comment]
Yeah lowering it worked for me, increased the number of documents and made them smaller.

Did you try saving the file and re-starting the python kernel for changes to take effect?

**Comment by WuYanZhao107** [comment]
> Yeah lowering it worked for me, increased the number of documents and made them smaller.
> 
> Did you try saving the file and re-starting the python kernel for changes to take effect? (it doesn't work if you don't restart the kernel after making the changes)

yeah, you are right. Thank you! I get more doucuments. So, what do you think which number is a good number for chunk? I've found that sometimes an entire paragraph will be linked to the first sentence of the next paragraph
